sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -4421,6 +4421,16 @@ object SQLConf {
       .booleanConf
       .createWithDefault(false)
 
+  val POSTGRES_GET_ARRAY_DIM_FROM_PG_ATTRIBUTE_METADATA_TABLE =
```

Comment:
We don't need this new parameter if pg's array_ndims satisfies all cases, we can completely remove the branch that visits the information_schema.

Reply:
I agree. I wanted to have the config just in case something goes wrong so we can fallback to previous behaviour.

Reply:
There is no 'previous/legacy behavior', this feature doesn't belong to any final official releases yet. There is nothing to restore. So, adding a legacy configuration is overkilling

Reply:
Thanks. Done.

Reply:
@yaooqinn may I get review again?